### ISSUE TITLE
[Bug]: Testing on swe-bench-lite stuck at success = self.ssh.prompt(timeout=timeout)

### ISSUE THREAD

**Issue by bin123apple** [OpenDevin/OpenDevin#2755]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

Hello! I am testing the swe-bench-lite dataset by using the` evaluation/swe_bench/scripts/run_infer.sh`  The container is successfully created and the llm is correctly generating outputs, however at the end of each instance, there is a get patch and evaluate step (`evaluation/swe_bench/run_infer.py`):

```
# ======= THIS IS SWE-Bench specific =======
    # Get git patch
    git_patch = sandbox.get_diff_patch()
    logger.info(f'Got git diff for instance {instance.instance_id}')
    # ==========================================

    # ======= Attempt to evaluate the agent's edits =======
    # TODO: if you need to do something in the sandbox to get the correctness metric, modify this function
    test_result = get_test_result(instance, sandbox, workspace_dir_name)
```

I don't know why the ssh_box always stuck at `git diff --no-color --cached b16c7d12ccbc7b2d20364b89fb44285bcbfede54`  This command is generated by this line of code in  `evaluation/swe_bench/swe_env_box.py  def get_diff_patch(self)`

```
# get the git diff
        exit_code, git_patch = self.execute(
            f'git diff --no-color --cached {self.swe_instance["base_commit"]}'
        )
```

In def execute() , it will stuck at this line,`success = self.ssh.prompt(timeout=timeout)`

Is there any suggestions? Thanks a lot for any help!

By the way, if I ssh to the docker, I can manually run this git diff command. But I will stuck by using the def execute() function outside of the docker. Thanks!

### Current OpenDevin version

```bash
OpenDevin Version 0.6.2

evaluation sandbox version: ghcr.io/opendevin/eval-swe-bench:full-v1.2.1
```

### Installation and Configuration

```bash
Start command: 

bash evaluation/swe_bench/scripts/run_infer.sh eval_gpt4o

config.toml

[core]
max_iterations = 100
cache_dir = "/tmp/cache"
sandbox_type = "ssh"
ssh_hostname = "localhost"
sandbox_timeout = 120

# SWEBench eval specific
use_host_network = false
run_as_devin = false
enable_auto_lint = true
max_budget_per_task = 4 # 4 USD

# TODO: Change these to the model you want to evaluate
[eval_gpt4o]
model = "gpt-4o"
api_key = "" # redacted
temperature = 0.0
```

### Model and Agent

Model: gpt-4o

Agent: codeact_swe_agent

### Operating System

System: Linux

### Reproduction Steps

Directly run: `bash evaluation/swe_bench/scripts/run_infer.sh eval_gpt4o`

It can finish create the docker container, generate the llm outputs. However, it will stuck at `git_patch = sandbox.get_diff_patch()
    logger.info(f'Got git diff for instance {instance.instance_id}'`

### Logs, Errors, Screenshots, and Additional Context

error log (the line number may not be accurate, because I added some debugging logs): 

```
21:56:54 - opendevin:ERROR: ssh_box.py:460 - Command "git diff --no-color --cached b16c7d12ccbc7b2d20364b89fb44285bcbfede54" timed out, killing process...
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/miniconda3/envs/data_engine/lib/python3.11/concurrent/futures/process.py", line 256, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/OpenDevin/evaluation/swe_bench/run_infer.py", line 340, in process_instance
    git_patch = sandbox.get_diff_patch()
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/OpenDevin/evaluation/swe_bench/swe_env_box.py", line 210, in get_diff_patch
    exit_code, git_patch = self.execute(
                           ^^^^^^^^^^^^^
  File "/home/OpenDevin/opendevin/runtime/docker/ssh_box.py", line 498, in execute
    return self._send_interrupt(cmd)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/OpenDevin/opendevin/runtime/docker/ssh_box.py", line 465, in _send_interrupt
    self.ssh.prompt()
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/pxssh.py", line 506, in prompt
    i = self.expect([self.PROMPT, TIMEOUT], timeout=timeout)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/spawnbase.py", line 354, in expect
    return self.expect_list(compiled_pattern_list,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/spawnbase.py", line 383, in expect_list
    return exp.expect_loop(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/expect.py", line 169, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/pty_spawn.py", line 500, in read_nonblocking
    if (timeout != 0) and select(timeout):
                          ^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/pty_spawn.py", line 450, in select
    return select_ignore_interrupts([self.child_fd], [], [], timeout)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/miniconda3/envs/data_engine/lib/python3.11/site-packages/pexpect/utils.py", line 143, in select_ignore_interrupts
    return select.select(iwtd, owtd, ewtd, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by bin123apple** [comment]
Thanks, this is exactly the reason,  the `ssh.prompt` will try to look for the `# (root user) or $ (other user)`. The `pager` setting will block it.  I added a new line : `exit_code, output = self.execute('git config --global core.pager ""')` before executing `git add -A`. I think it should be good now. Just submitted a PR # https://github.com/OpenDevin/OpenDevin/pull/2828.  Thanks!

**Comment by SmartManoj** [comment]
The issue is solved by #2828
